go-controller/pkg/ovn/gateway_init.go

```diff
@@ -421,8 +423,13 @@ func addDistributedGWPort() error {
 		return nil
 	}
 
-	datapath, stderr, err := util.RunOVNSbctl("--data=bare", "--no-heading", "--columns=_uuid", "find", "datapath",
-		"external_ids:name="+ovnClusterRouter)
+	// Wait a bit for northd to create the cluster router's datapath in southbound
+	var datapath string
+	err = wait.PollImmediate(500*time.Millisecond, 30*time.Second, func() (bool, error) {
+		datapath, stderr, err = util.RunOVNSbctl("--data=bare", "--no-heading", "--columns=_uuid", "find", "datapath",
+			"external_ids:name="+ovnClusterRouter)
+		return datapath != "", err
```

Comment:
should this be `return datapath != "", nil`

Reply:
does it errors if the datapath is not found?

Reply:
> does it errors if the datapath is not found?

@aojea no, it doesn't:

```
$ ovn-sbctl --data=bare --no-heading --columns=_uuid find datapath external_ids:name=dsafdsaf
$ echo $?
0
```
but yeah, if there's some other error with sbctl I suppose we should just retry.

Reply:
changed.